pydicom/sequence.py

```diff
@@ -91,29 +116,34 @@ def parent(self, value: Dataset) -> None:
             for item in self._list:
                 item.parent = self._parent
 
-    @overload
-    def __setitem__(self, i: int, val: Dataset) -> None: pass
+    @overload  # type: ignore[override]
+    def __setitem__(self, idx: int, val: Dataset) -> None: pass
 
     @overload
-    def __setitem__(self, i: slice, val: SeqType[Dataset]) -> None: pass
+    def __setitem__(self, idx: slice, val: Iterable[Dataset]) -> None: pass
 
     def __setitem__(
-        self, i: Union[slice, int], val: Union[SeqType[Dataset], Dataset]
+        self, idx: Union[slice, int], val: Union[Iterable[Dataset], Dataset]
     ) -> None:
         """Set the parent :class:`~pydicom.dataset.Dataset` to the new
         :class:`Sequence` item
         """
-        super().__setitem__(i, val)
-        if isinstance(i, slice):
+        if isinstance(idx, slice):
+            if isinstance(val, Dataset):
```

Comment:
I added this because the exception message wasn't really confusing `Sequence contents must be Dataset instances.`